system/doc/reference_manual/data_types.xml

```diff
@@ -79,6 +79,25 @@
 11> <input>1_234.333_333</input>
 1234.333333
 </pre>
+<marker id="float_represenation_problem" /><note>
+  <p>Float literals cannot always be
```

Comment:
Besides the confusion about floating point numbers not being
represented exactly, there is also confusion about floating point
numbers not being printed as expected. Typically it's about numbers
such as float(1 bsl 53), which is indeed represented exactly, but has
the same representation as, for example, float(1 bsl 53 + 1).

I think all basic information can be found in
https://bugs.erlang.org/browse/ERL-1308; the problem is to explain it
correctly and succinctly so that everyone can understand.

Reply:
`1 bsl 53' is not a good example. One needs to go up to `1 bsl 55'.

Reply:
@uabboli I have added a comment to https://bugs.erlang.org/browse/ERL-1308 . Can you have a look? Can you also have a look at @garazdawi's suggestion for improving this above and let us know if you think it explains the "not being printed as expected" part in a satisfactory way?

Reply:
Interesting that Python's output differs. I've always thought Scheme does everything right when it comes to printing floating point numbers, so I downloaded mit-scheme: they print Lukas' example like Erlang does. It seems Python is using a modified algorithm.

BTW, there is another example where a zero can be misunderstood: the syntax of Erlang's floating point numbers requires a digit before and after the dot, while Scheme's syntax does not, which is to their advantage.

Regarding Lukas' suggestion. I find it a bit too succinct. Given the example '36028797018963969.0.', how am I supposed to think? Isn't it possible, or even likely, that I write a bug report?

Reply:
There is a spelling error in the marker (line 82)
<marker id="float_represenation_problem" /><note>
  <p>Float literals cannot always be

Should be <marker id="float_representation_problem" />....

Reply:
My thinking was that we rely on external pages to do the in-depth description of why floating points behave the way they do and just have a basic description of the problem in our documentation. We could of course have a larger better description, like for instance [elixir](https://hexdocs.pm/elixir/Float.html#module-known-issues).

Reply:
Elixir's explanation does not mention the rounding behavior of the pretty-printer, nor does the external paper. You refer to "the problem", but I claim there are two "problems", the other one being the pretty-printer algorithm. Both problems seem to generate bug reports.

Reply:
I have looked into this and I think it makes sense to change the printing algorithm so the example becomes:

```
> 36028797018963969.0.
3.602879701896397e16
```

instead of:

```
> 36028797018963969.0.
36028797018963970.0
```

This should make the example less confusing. I will add a commit with this change and an attempt to do a better explanation next week.

Reply:
I have changed the printing algorithm so that 36028797018963969.0 is printed as 3.602879701896397e16 and updated the documentation accordingly. See 09bc530